cli/lib/commands/init.js

```diff
@@ -0,0 +1,210 @@
+ */
+function validateName(name) {
+  // TODO: Uese Regex for "name@registry-domain.dev/package_name"?
+  return name && name.length > 0;
+}
+
+/**
+ * Validates the version provided by the user
+ */
+function validateVersion(version) {
+  return isValidSemver(version);
+}
+
+/**
+ * Used to provide consistent casing for hash lookups
+ */
+function lowercase(str) {
+  return str.toLowerCase();
+}
+
+/**
+ * Minimal prompt for asking users for input
+ */
+function ask(question, rl) {
+  return new Promise((resolve, reject) => {
+    rl.question(question, answer => resolve(answer));
+  }).catch(e => {
+    console.error(e);
+  });
+}
+
+/**
+ * Asks user a question until a valid answer is given and applies any
+ * desired transformations to the value returned if provided
+ */
+async function askQuestion(question, rl, validator, transform = undefined) {
+  let invalid = true;
+  let ans = undefined;
+  const { MESSAGE, REQUIRED } = question;
+
+  while (invalid) {
+    ans = await ask(MESSAGE, rl);
+
+    if (transform) {
+      ans = transform(ans);
+    }
+
+    if (ans.length < 1 && REQUIRED) {
+      console.error(REJECTION_MSGS.REQUIRED);
+      continue;
+    }
+
+    if (!validator || validator(ans)) {
+      invalid = false;
+    } else {
+      console.error(REJECTION_MSGS[validator.name]);
+    }
+  }
+
+  return ans;
+}
+
+/**
+ * Exported function used as `ds create`
+ */
+async function create(opts) {
+  await writeFile(`${process.cwd()}/package.json`, createPackageJson());
+
+  try {
+    const rl = readline.createInterface({
+      input: process.stdin,
+      output: process.stdout
+    });
+
+    if (fileExists(tomlLocation())) {
+      // Ask if we should proceed since the Package.toml exits
+      const proceed = await askQuestion(
+        QUESTIONS.TOML_EXISTS,
+        rl,
+        validateYesNo,
+        lowercase
+      );
+
+      if (VALID_YES_NO[proceed] === 0) {
+        console.log('Exiting.');
+        rl.close();
+        return 0;
+      }
+    }
+
+    let answers = [];
+
+    answers['name'] = await askQuestion(QUESTIONS.NAME, rl, validateName);
+    answers['version'] = await askQuestion(
+      QUESTIONS.VERSION,
+      rl,
+      validateVersion
+    );
+    answers['entry'] = await askQuestion(QUESTIONS.ENTRY, rl);
+    answers['type'] = await askQuestion(QUESTIONS.TYPE, rl);
+    answers['license'] = await askQuestion(QUESTIONS.LICENSE, rl);
+    answers['description'] = await askQuestion(QUESTIONS.DESCRIPTION, rl);
+    answers['homepage'] = await askQuestion(QUESTIONS.HOMEPAGE, rl);
+    answers['author'] = await askQuestion(QUESTIONS.AUTHOR, rl);
+    answers['repository'] = await askQuestion(QUESTIONS.REPOSITORY, rl);
+
+    rl.close();
+
+    await writeFile(tomlLocation(), createToml(answers));
+
+    if (answers['type'] === 'module') {
+      await writeFile(process.cwd(), createPackageJson());
+    }
+  } catch (e) {
+    console.error('There was an error creating your Package.toml');
```

Comment:
Avoid `try / catch` that covers too much of code. Operation specific errors are generally a lot easier to digest and work with.